train.rb

```diff
@@ -0,0 +1,68 @@
+class Train
+  attr_reader :velocity, :carriage_num, :type
+
+  def initialize(id, type, carriage_num)
+    @id = id
+    @type = type
+    @carriage_num = carriage_num
+
+    @velocity = 0
+  end
+
+  def accelerate_by(delta_velocity)
+    @velocity += delta_velocity
+  end
+
+  def decrease_speed(delta_velocity)
+    @velocity -= delta_velocity
+  end
+
+  def add_carriage
+    if @velocity != 0
+      @carriage_num += 1
+    end
+  end
+
+  def remove_carriage
+    if @velocity != 0 and @carriage_num != 0
+      @carriage_num -= 1
+    end
+  end
+
+  def set_route(route)
+    @route = route
+    @cur_station_index = 0
```

Comment:
```suggestion
    @current_station_index = 0
```
экономия 4-х символов особо не помогает, а читаемость падает